File: backend/app/schemas/role.py
# app/schemas/role.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class PermissionBase(BaseModel):
    id: int
    name: str
    display_name: str

    class Config:
        from_attributes = True

class RoleBase(BaseModel):
    name: str = Field(..., max_length=50)
    display_name: str = Field(..., max_length=100)
    description: Optional[str] = None

class RoleCreate(RoleBase):
    is_active: bool = True
    permission_ids: Optional[List[int]] = []

class RoleUpdate(BaseModel):
    display_name: Optional[str] = Field(None, max_length=100)
    description: Optional[str] = None
    is_active: Optional[bool] = None
    permission_ids: Optional[List[int]] = None

class RoleResponse(RoleBase):
    id: int
    is_system_role: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime
    permissions: List[PermissionBase] = []

    class Config:
        from_attributes = True

File: frontend/lib/app/models/role.dart
import 'package:json_annotation/json_annotation.dart';
import 'permission.dart';

part 'role.g.dart';

@JsonSerializable()
class Role {
  final int id;
  @JsonKey(defaultValue: '')
  final String name;
  @JsonKey(name: 'display_name', defaultValue: '')
  final String displayName;
  final String? description;
  @JsonKey(name: 'is_system_role', defaultValue: false)
  final bool isSystemRole;
  @JsonKey(name: 'is_active', defaultValue: true)
  final bool isActive;
  @JsonKey(name: 'created_at')
  final DateTime? createdAt;
  @JsonKey(name: 'updated_at')
  final DateTime? updatedAt;
  @JsonKey(defaultValue: [])
  final List<Permission> permissions;

  Role({
    required this.id,
    this.name = '',
    this.displayName = '',
    this.description,
    this.isSystemRole = false,
    this.isActive = true,
    this.createdAt,
    this.updatedAt,
    this.permissions = const [],
  });

  factory Role.fromJson(Map<String, dynamic> json) => _$RoleFromJson(json);

  Map<String, dynamic> toJson() => _$RoleToJson(this);
}
